### DISCUSSION TITLE
RuntimeError: Native API failed. Native API returns: -6 (PI_ERROR_OUT_OF_HOST_MEMORY) -6 (PI_ERROR_OUT_OF_HOST_MEMORY)

### DISCUSSION THREAD

**Discussion by Whackjob** [vladmandic/sdnext#2326] [General]
I've been troubleshooting a lot of issues, but I'm at the point where I can almost get SD.Next to run perfectly.  It's slower than it should be (based on a docker I got working once, briefly, that ran beautifully, but couldn't replicate a successful run) but I'm in.  On load all errors are now gone except cuda related ones.  I'm using an Intel Arc 770 on a Linux Mint system with the latest stable kernel.

Anyway.  A regular image, no loras, here's my stats on a test run:

```_Steps: 20 | Seed: 1623540145 | Sampler: Default | CFG scale: 6 | Size: 1024x1024 | Parser: Fixed attention | Model: XL_lunaMia_v50bakedvae | Model hash: f411b79977 | Backend: Diffusers | Version: 98bf24d | Operations: txt2img | Token merging ratio: 0.4 | CFG rescale: 0.7

Time: 45.07s | **GPU active 16062 MB reserved 16062** | used 322 MB free 15967 MB total 16288 MB | retries 0 oom 0_```

Please note the bold. When that docker worked, I'd use maybe 7GB of VRAM.  For some reason, now it says all is used up?  But the real magic is when I try to use a LORA.  As so, this first part as it loads:

```_23:34:00-661913 DEBUG    Applying token merging: ratio=0.4                      
23:34:00-786038 ERROR    LoRA loading failed: TestLora loader="diffusers"
                         The following keys have not been correctly be renamed: 
23:34:00-790363 DEBUG    Diffuser pipeline: StableDiffusionXLPipeline_   ```

But I do get an image, maybe two.  Eventually, I MIGHT get this error and every attempt after fails:

```RuntimeError: Native API failed. Native API returns: -6 (PI_ERROR_OUT_OF_HOST_MEMORY) -6 (PI_ERROR_OUT_OF_HOST_MEMORY)```

If I don't get that, what happens is, I figure, all my VRAM gets taken up, my screens freeze, and I gotta cold reboot.

I think maybe part of the trouble might be using diffusers with lora, but I get the VRAM being taken at 100% even before I use those.

Could I get some assistance, please?  It really feels like I'm almost finally squared away.

**Comment by Disty0** [comment]
You were using CPU before this run. Remove config.json and let WebUI create new config for GPU.

**Reply by Whackjob** [reply]
Thank you!  I'm about 95% of the way there, now.  After doing that, things did work great for a good while.  But before I went to sleep last night, I queued up a run of 50 images, but just one at a time.  I woke up to the same error, again.  I get this error on the startup, now, does this perhaps explain why?

```07:58:05-531337 ERROR    Loading diffusers VAE failed:                          
                         /media/whackjob/16Tons/stable-diffusion/models/VAE/sdxl
                         _vae.safetensors Error while deserializing header:     
                         HeaderTooLarge   ```

**Reply by Disty0** [reply]
Run MemTest to check for RAM errors. You are getting too many corrupted file related errors in a row.

**Reply by Whackjob** [reply]
Brother, you've hit the nail on the head.  I pulled down MemTest86.  Took 11 hours to do an in-depth test for the 64GB ram I had.  Well, test result was a big 'ol FAIL.  Thinking on it, I had bought two 16GB sticks, and a buddy of mine had a spare two that he pulled from something-or-other.  I figure, those are the most likely bad sticks.  So, I pulled 'em.  I haven't run another test of the original ram, I'll do that tonight when I go to sleep.

Tried SD again, immediately got memory errors.  Looking at it, it was obvious it was trying to use RAM that wasn't there.  So I deleted the config file and I've got it running up again and redoing the configs.

**Reply by Disty0** [reply]
It's probably a good idea to reinstall your OS if you had a bad memory.

**Reply by Whackjob** [reply]
I finally got your tip to you last night.  Hopefully it was worth your time.  I do appreciate all the help you've given me.
